### ISSUE TITLE
Fix accessibility issue for links opening in a new tab

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2093]
## Bug Description

Any links that open in a new tab should be marked with an "external link" icon (which mostly is the case now already), but also with a screen reader-only text that says "(opens in a new tab)". Related: #1684 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All external links in Site Kit should be annotated with an `aria-label` attribute containing `_x( '(opens in a new tab)', 'screen reader text', 'google-site-kit' )`
	* An external link that receives an `aria-label` attribute should have the new tab text appended to it
	* An external link that does not receive a `aria-label` attribute and has string `children` should append the new tab text to the `children` for the `aria-label` attribute
* Any existing links that use the `<span class="screen-reader-text">` element in the children to truncate the link text between visible and hidden portions should be updated to use separate translation strings where the full version is used as the `aria-label` and previously truncated version as the children

## Implementation Brief

* Update `assets/js/components/link.js` to set an `aria-label` attribute based on the logic in the AC
* Update usage of `Link` component with `<span class="screen-reader-text">` to use an `aria-label` attribute instead

<details>
<summary>Original IB</summary>

In the `link (https://github.com/google/site-kit-wp/blob/develop/assets/js/components/link.js)` component (`assets/js/components/link.js`):

- Add the following WordPress dependencies:
  - `import { _x } from '@wordpress/i18n';`
  - `import { VisuallyHidden } from '@wordpress/components';`

- In the render function, after outputting the { children } , conditionally display the "(opens in a new tab)" string. 
`
{ external && (
					<VisuallyHidden as="span" className="screen-reader-text">
						{ _x( '(opens in a new tab)', 'screen reader text', 'google-site-kit' ) }
					</VisuallyHidden>
				) }
`

</details>

## QA Brief
- Inspect any external link using Chrome devtools. They are easily recognisable by having the external icons.
- The anchor tag should have the aria-label attribute with the value of the text within the link and the following text: "(opens in a new tab)"
- No visual regressions should be noted.

## Changelog entry

* Fix accessibility issue with links opening in a new tab by annotating them with screen reader text informing about it.

**Comment by aaemnnosttv** [comment]
IB looks good for the most part, although I think it would be a good idea to use a dedicated component for screen reader text. Gutenberg has a `VisuallyHidden` component in `@wordpress/components` which is really flexible. It would be great if we could use that.

**Comment by asvinb** [comment]
Thanks @aaemnnosttv . IB updated.

**Comment by tofumatt** [comment]
IB ✅

I think this is a fine stop-gap, but as #1684 mentions: the main issue here isn't that the "new tab" links aren't outlined, it's that they can't be controlled by the end-user.

Still, this is an improvement and is straightforward to implement, so 👍🏻

**Comment by tofumatt** [comment]
(Note that this probably isn't perfect from an i18n perspective, because the order of the children and the parenthetical isn't combined, but again: as this is a stop-gap I think it's okay.)

**Comment by aaemnnosttv** [comment]
@felixarntz @tofumatt I've updated the ACs here to use the `aria-label` attribute which is better suited to solving this problem and doesn't restrict us from using things like `dangerouslySetInnerHTML` (and even eliminates the need for us to use this at all in many cases).

For more info about this see: https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA8

**Comment by tofumatt** [comment]
👍🏻 Good call on the use of `aria-label` here, that's a better solution. Revised AC/IB ✅

**Comment by ryanwelcher** [comment]
Manually inspected each external link in the admin:

**Dashboard Page:**
PageSpeed Insights link in Page Speed and Experience is missing the `aria-label`
![missing-aria-label](https://user-images.githubusercontent.com/1259027/98128494-013a9700-1e86-11eb-9d58-a2f6aba4ebd3.png)

**Search Console:** Passed
**AdSense:** Passed
**Analytics:** Passed
**Settings:** Passed

Sending back to Execution

**Comment by aaemnnosttv** [comment]
Nice catch @ryanwelcher 👍 

Here's another one, which oddly isn't even using the `Link` component where it should be:
https://github.com/google/site-kit-wp/blob/5ace4c0ddd772de2785237acaf83173169baf4bd/assets/js/modules/analytics/components/settings/SettingsView.js#L90-L97

As for the first one, this can use `createInterpolateElement` to use the real `Link` component now as well.
https://github.com/google/site-kit-wp/blob/5ace4c0ddd772de2785237acaf83173169baf4bd/assets/js/modules/pagespeed-insights/components/common/ReportDetailsLink.js#L40

**Comment by tofumatt** [comment]
QA ✅ Looks good now, thanks! 👍🏻
